cartographer_ros/cartographer_ros/map_builder_bridge.cc

```diff
@@ -177,20 +193,23 @@ visualization_msgs::MarkerArray MapBuilderBridge::GetTrajectoryNodeList() {
        trajectory_id < static_cast<int>(all_trajectory_nodes.size());
        ++trajectory_id) {
     const auto& single_trajectory_nodes = all_trajectory_nodes[trajectory_id];
-    visualization_msgs::Marker marker;
-    marker.ns = "Trajectory " + std::to_string(trajectory_id);
-    marker.id = 0;
-    marker.type = visualization_msgs::Marker::LINE_STRIP;
-    marker.header.stamp = ::ros::Time::now();
-    marker.header.frame_id = node_options_.map_frame;
-    marker.color = ToMessage(cartographer::io::GetColor(trajectory_id));
-    marker.scale.x = kTrajectoryLineStripMarkerScale;
-    marker.pose.orientation.w = 1.0;
-    marker.pose.position.z = 0.05;
+    bool jump = false;
+    int num_splits = 0;
+    auto marker = CreateTrajectoryMarker(trajectory_id, num_splits,
+                                         node_options_.map_frame);
+
     for (const auto& node : single_trajectory_nodes) {
       if (node.trimmed()) {
+        jump = true;
```

Comment:
Similar to the handling large markers: Push the marker here if it contains a line and always reset it? Then we do not need the `jump` variable.

Reply:
`jump` is necessary. It would push empty markers as many as trimmed nodes if we push marker here.

Reply:
That's what I meant by "if it contains a line", i.e. at least two points. Maybe even extract a function to do the `PushAndResetLineMarker()` and use it in all three places.

Reply:
ah yes you are right.